src/otaclient/ota_core.py

```diff
@@ -536,40 +545,45 @@ def _execute_update(self):
 
         # ------ pre-update ------ #
         logger.info("enter local OTA update...")
+
+        with TemporaryDirectory() as _tmp_dir:
+            use_inplace_mode = can_use_in_place_mode(
+                dev=self._boot_controller.standby_slot_dev,
+                mnt_point=_tmp_dir,
+                threshold_in_bytes=int(
+                    self._ota_metadata.total_regulars_size
+                    * STANDBY_SLOT_USED_SIZE_THRESHOLD
+                ),
+            )
+        logger.info(
+            f"check if we can use in-place mode to update standby slot: {use_inplace_mode}"
+        )
+
         self._boot_controller.pre_update(
             self.update_version,
-            standby_as_ref=False,  # NOTE: this option is deprecated and not used by bootcontroller
-            erase_standby=True,  # NOTE: as of now, we only have rebuild mode, so always erase_standby
+            # NOTE: this option is deprecated and not used by bootcontroller
+            # TODO:(20250613) when standby_as_ref is set, skip mounting active slot.
+            #       we cannot do this for now, as some boot controller impl still refer to
+            #       active_slot mounts.
+            standby_as_ref=use_inplace_mode,
+            erase_standby=not use_inplace_mode,
         )
-
         # prepare the tmp storage on standby slot after boot_controller.pre_update finished
-        self._resource_dir_on_standby.mkdir(exist_ok=True)
+        self._resource_dir_on_standby.mkdir(exist_ok=True, parents=True)
+        self._ota_tmp_meta_on_standby.mkdir(exist_ok=True, parents=True)
```

Comment:
NOTE: for inplace update mode, these two OTA working folders will be ignored by inplace update mode delta generator.
Also see create_standby.delta_gen for more details.

Reply:
How about rebuild mode? Just will be ignored like the following?
1. create directory
2. calculated delta based on active
3. copy active to standby, and remove this directory